test/core_tests.js

```diff
@@ -0,0 +1,138 @@
+QUnit.begin(function( details ) {
+  // In case tests didn't complete, we are left with a hash/anchor pointing somewhere. But we want to start from scratch:
+  window.location.hash = "";    
+});
+
+QUnit.test( "Initialize Impress.js", function( assert ) {
+  assert.ok( impress, 
+             "impress declared in global scope" );
+  assert.strictEqual( impress().init(), undefined,
+                    "impress().init() called." );
+  assert.strictEqual( impress().init(), undefined,
+                    "It's ok to call impress().init() a second time, it's a no-op." );
+  var notSupportedClass = document.body.classList.contains("impress-not-supported");
+  var yesSupportedClass = document.body.classList.contains("impress-supported");
+  if ( !_impressSupported() ) {
+    assert.ok( notSupportedClass,
+               "body.impress-not-supported class still there." );
+    assert.ok( !yesSupportedClass,
+               "body.impress-supported class was NOT added." );
+  } else {
+    assert.ok( !notSupportedClass,
+               "body.impress-not-supported class was removed." );
+    assert.ok( yesSupportedClass,
+               "body.impress-supported class was added." );
```

Comment:
Now thinking again about this, does it makes sense testing that the `.not-supported` class is being added to the body element and vice-versa? I mean, we only run this test manually for now, and probably in evergreen browsers that support impress.js in a decent stability (Chrome for example).

Ideally this sort of test should be assertive on what it expects (or do not expect) to happen without caring about the environment it is running. If necessary, we should simulate an environment to make impress.js trigger the secondary condition to test if the `.not supported` is being properly executed, otherwise it is not going to run in every run and therefore we are not going to catch it (only if we run in a different browser that specifically doesn't support impress.js).

I suggest just asserting that it adds the class if it is supported and then later find a way (stubbing or mocking some data) to assert that the `.not-supported` class was added correctly.